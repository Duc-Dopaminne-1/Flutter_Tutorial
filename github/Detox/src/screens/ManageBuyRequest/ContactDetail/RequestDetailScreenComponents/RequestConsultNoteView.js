import React, {useState} from 'react';
import {StyleSheet, TextInput, View} from 'react-native';

import {SIZES} from '../../../../assets/constants/sizes';
import {translate} from '../../../../assets/localize';
import {STRINGS} from '../../../../assets/localize/string';
import {COLORS} from '../../../../assets/theme/colors';
import {FONTS} from '../../../../assets/theme/fonts';
import {METRICS} from '../../../../assets/theme/metric';
import {commonStyles} from '../../../../assets/theme/styles';
import TextView from '../../../../components/TextView';

const styles = StyleSheet.create({
  textHeader1: {
    ...FONTS.bold,
    ...FONTS.fontSize18,
  },
  consultNoteInput: {
    width: '100%',
    height: 80,
    borderWidth: SIZES.BORDER_WIDTH_1,
    borderColor: COLORS.GREY_E4,
    ...METRICS.smallPadding,
  },
});

const RequestConsultNoteView = ({onChangeText}) => {
  const [notes, setNotes] = useState('');

  const SYMBOL_COLON = ':';
  const onChangeTextNotes = text => {
    setNotes(text);
    onChangeText(text);
  };

  return (
    <>
      <TextView title={translate(STRINGS.CONSULT_NOTE) + SYMBOL_COLON} value="" />
      <View style={commonStyles.separatorRow8} />
      <TextInput
        style={styles.consultNoteInput}
        multiline
        onChangeText={onChangeTextNotes}
        value={notes}
      />
    </>
  );
};

export default RequestConsultNoteView;
