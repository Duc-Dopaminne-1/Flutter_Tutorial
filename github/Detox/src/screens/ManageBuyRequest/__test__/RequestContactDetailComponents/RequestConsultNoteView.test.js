import React from 'react';
import renderer from 'react-test-renderer';

import RequestConsultNoteView from '../../ContactDetail/RequestDetailScreenComponents/RequestConsultNoteView';

it('renders correctly', () => {
  const tree = renderer.create(<RequestConsultNoteView />).toJSON();
  expect(tree).toMatchSnapshot();
});
