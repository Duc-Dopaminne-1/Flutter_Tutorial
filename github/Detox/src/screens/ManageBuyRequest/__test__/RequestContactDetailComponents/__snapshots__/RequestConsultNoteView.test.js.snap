// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#A3A3A3",
            "fontFamily": "Nunito-Regular",
            "fontSize": 14,
            "includeFontPadding": true,
          },
          Object {},
        ]
      }
    >
      Ghi chú tư vấn:
    </Text>
    <View
      style={
        Array [
          Object {
            "width": 16,
          },
          Object {
            "width": 16,
          },
        ]
      }
    />
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "Nunito-Regular",
            "fontSize": 14,
            "includeFontPadding": true,
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Object {
        "height": 8,
      }
    }
  />,
  <TextInput
    multiline={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "#E4E4E4",
        "borderWidth": 1,
        "height": 80,
        "padding": 8,
        "width": "100%",
      }
    }
    value=""
  />,
]
`;
